autonomy/analyse/abci/app_spec.py

```diff
@@ -57,6 +87,107 @@ class OutputFormats:  # pylint: disable=too-few-public-methods
         MERMAID = "mermaid"
         ALL = (JSON, YAML, MERMAID)
 
+        default_output_files = {
+            JSON: DEFAULT_FSM_SPEC_JSON,
+            YAML: DEFAULT_FSM_SPEC_YAML,
+            MERMAID: DEFAULT_FSM_SPEC_MERMAID,
+        }
+
+    @classmethod
+    def from_yaml(cls, file: Path) -> Dict:
+        """Load from yaml."""
+        with open_file(file, mode="r", encoding="utf-8") as fp:
+            data = yaml.safe_load(fp)
+
+        validate_fsm_spec(data)
+        return data
```

Comment:
- This is not a classmethod. Class methods are those that alter the class or those that return an instance of the class.
- if validate_fsm_spec were to return its input
`return validate_fsm_spec(yaml.safe_load(file.read_text(encoding="utf-8")))`

Reply:
I'm sorry for not making my intent clear with this one, I made this class method because I wanted to make the validation method a part of this class after we move the schema to a json file. If we keep the validation method outside the class the we will have to load it everytime a load/dump method get's called. Once I move the schema to a json file I'll define the validation method as a part of the loader class and make the validator a class variable. It'll make much more sense after that